cmd/admin-bucket-quota.go

```diff
@@ -0,0 +1,188 @@
+	q.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(q, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+var adminBucketQuotaCmd = cli.Command{
+	Name:   "quota",
+	Usage:  "Manage bucket quota",
+	Action: mainAdminBucketQuota,
+	Before: setGlobalsFromContext,
+	Flags:  append(adminQuotaFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+ 
+USAGE:
+   {{.HelpName}} TARGET [--fifo QUOTA | --hard QUOTA | --clear]
+ 
+QUOTA
+   quota accepts human-readable case-insensitive number
+   suffixes such as "k", "m", "g" and "t" referring to the metric units KB,
+   MB, GB and TB respectively. Adding an "i" to these prefixes, uses the IEC
+   units, so that "gi" refers to "gibibyte" or "GiB". A "b" at the end is
+   also accepted. Without suffixes the unit is bytes.
+ 
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
+   1. Display bucket quota configured for "mybucket" on MinIO.
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket
+	
+   2. Set FIFO quota for a bucket "mybucket" on MinIO.
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket --fifo 64kB
+
+   3. Set hard quota of 1gb for a bucket "mybucket" on MinIO.
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket --hard 1gb
+
+   4. Clear bucket quota configured for bucket "mybucket" on MinIO.
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket --clear
+
+`,
+}
+
+// checkAdminBucketQuotaSyntax - validate all the passed arguments
+func checkAdminBucketQuotaSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelpAndExit(ctx, "quota", 1) // last argument is exit code
+	}
+
+	if ctx.IsSet("hard") && ctx.IsSet("fifo") {
+		fatalIf(errInvalidArgument(), "Only one of --hard or --fifo flags can be set")
+	}
+	if (ctx.IsSet("hard") || ctx.IsSet("fifo")) && len(ctx.Args()) == 0 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args()...), "please specify bucket and quota")
+	}
+	if ctx.IsSet("clear") && len(ctx.Args()) == 0 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args()...), "clear flag must be passed with target alone")
+	}
+}
+
+// mainAdminBucketQuota is the handler for "mc admin bucket quota" command.
+func mainAdminBucketQuota(ctx *cli.Context) error {
+	checkAdminBucketQuotaSyntax(ctx)
+
+	console.SetColor("QuotaMessage", color.New(color.FgGreen))
+	console.SetColor("QuotaInfo", color.New(color.FgBlue))
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+	quotaStr := ctx.String("fifo")
+	if ctx.IsSet("hard") {
+		quotaStr = ctx.String("hard")
+	}
+	_, targetURL := url2Alias(args[0])
+	if ctx.IsSet("fifo") || ctx.IsSet("hard") && len(args) == 1 {
+		qType := madmin.FIFOQuota
+		if ctx.IsSet("hard") {
+			qType = madmin.HardQuota
+		}
+		quota, err := humanize.ParseBytes(quotaStr)
```

Comment:
err should be `e` in `mc` Go error should be `e` and probe Error should be `err`